roxie/ccd/ccdserver.cpp

```diff
@@ -11273,10 +11270,25 @@ class CRoxieServerXmlWriteActivity : public CRoxieServerDiskWriteActivity
         desc->queryProperties().setProp("@format","utf8n");
         desc->queryProperties().setProp("@rowTag",rowTag.get());
         desc->queryProperties().setProp("@kind", (kind==TAKjsonwrite) ? "json" : "xml");
-        if (headerLength)
-            desc->queryProperties().setPropInt("@headerLength", headerLength);
-        if (footerLength)
-            desc->queryProperties().setPropInt("@footerLength", footerLength);
+        if (kind==TAKjsonwrite)
+        {
+            if (headerLength)
+                desc->queryProperties().setPropInt("@headerLength", headerLength);
+            if (footerLength)
+                desc->queryProperties().setPropInt("@footerLength", footerLength);
+        }
+        else
+        {
+            IPartDescriptorIterator *parts = desc->getIterator();
+            if (parts->first())
+            {
+                do {
+                    IPartDescriptor &part = parts->query();
+                    part.queryProperties().setPropInt("@headerLength", headerLength);
+                    part.queryProperties().setPropInt("@footerLength", footerLength);
```

Comment:
I see the added comment:
```
// If multiple XML files with different header/footer sprayed into a
// multi slaves Thor system then every parts can have different length
// header/footer. For compatibility reason Roxie does same.
```

But I'm still dubious this change (setting per Part) is correct here (and in similar change in hthor/thor code).
Spray might be spraying files with different header/footer lengths, but here in the engines, all header/footer lengths must be same for each part.

I see code elsewhere reading headerLength at the File level.
The one place that reads it at the Part level, FileSprayer::analyseFileHeaders, would appear to also read it at the File level and use it as the default (if nothing specified at the part level).

So I would have thought this code in the engines should just set at the File level.

@AttilaVamos - can you explain why not?